src/java/org/apache/cassandra/repair/autorepair/RepairTokenRangeSplitter.java

```diff
@@ -61,163 +63,258 @@
+ * To manage these issues, the strategy involves estimating the size and number of partitions within a range and
+ * splitting it accordingly to bound the size of the range splits.  This is established by iterating over SSTable
+ * index files to estimate the amount of bytes and partitions involved in the ranges being repaired and by what
+ * repair type is beinb invoked.
+ * <p/>
+ * While this splitter has a lot of tuning parameters, the expectation is that the established default configuration
+ * shall be sensible for all {@link org.apache.cassandra.repair.autorepair.AutoRepairConfig.RepairType}'s. The following
+ * configuration parameters are offered.
+ * <ul>
+ *     <li>
+ *         <b>bytes_per_assigment</b>: The target and maximum amount of bytes that should be included in a repair
+ *         assignment. This is meant to scope the amount of work involved in a repair.  For incremental repair, this
+ *         involves the total number of bytes in all SSTables containing unrepaired data involving the ranges being
+ *         repaired, including data that doesn't cover the range.  This is to account for the amount of anticompaction
+ *         that is expected. For all other repair types, this involves the amount of data covering the range being
+ *         repaired.
+ *     </li>
+ *     <li>
+ *         <b>partitions_per_assignment</b>: The target number of partitions that should be included in a repair
+ *         assignment.  This configuration exists to reduce excessive overstreaming.
+ *     </li>
+ *     <li>
+ *         <b>max_tables_per_assignment</b>: The maximum number of tables that can be included in a repair assignment.
+ *         This aims to reduce the number of repairs, especially in cases where a large amount of tables exists for
+ *         a keyspace.  Note that the splitter will avoid batching tables together if they exceed the other
+ *         configuration paramters such as <code>bytes_per_assignment</code> and <code>partitions_per_assignment</code>.
+ *     </li>
+ *     <li>
+ *         <b>max_bytes_per_schedule</b>: The maximum number of bytes to cover an individiual schedule.  This serves
+ *         as a mechanism for throttling the amount of work that can be done on each repair cycle.  One may opt to
+ *         reduce this value if the impact of repairs is causing too many load on the cluster, or increase it if
+ *         writes outpace the amount of data being repaired.  Alternatively, one may want to choose tuning down or up
+ *         the <code>min_repair_interval</code>.
+ *     </li>
+ * </ul>
+ * Given the impact of what each repair type accomplishes, different defaults are established per repair type.
+ * <ul>
+ *     <li>
+ *         <b>full</b>:  Configured in a way that attempts to accomplish repairing all data in a schedule, with
+ *         individual repairs targeting at most 200GiB of data and 1048576 partitions.
+ *         <ul>
+ *             <li><b>bytes_per_assignment</b>: 200GiB</li>
```

Comment:
Thoughts on this?  My original rationale was that the `partitions_per_assignment` guardrail would do well to prevent over-streaming, but allowing ~200GB validations might be a bit much.

Although, this doesn't factor in compression, so if you have ~2TB nodes and 50% compression, it'd likely create 20 splits, so seems about right.

Intending to try this out in practice this weekend.